include/deal.II/base/exceptions.h

```diff
@@ -1228,6 +1228,23 @@ namespace StandardExceptions
                                            dealii::ExcCudaError(cudaGetErrorString(error_code)))
 #endif
 
+#if defined(DEAL_II_WITH_CXX11) || !defined(DEAL_II_MSVC)
```

Comment:
I think this should only test for C++11. In those (few) places where MSVC doesn't work, we should leave the `#ifndef MSVC` in place, but in all other places we can at least get rid of the check for C++11.